Cross platform incompatible parts fixed

kivancgnlp opened the pull request:
Tested on MacOS and Ubuntu Linux.  There still some parts need modification for MacOS.

kivancgnlp committed:
Cross platform incompatible parts fixed

kivancgnlp committed:
Cross platform incompatible parts fixed

anjaldoshi commented:
Hi @kivancgnlp,

Thank you for submitting this PR.

I wanted to get some clarity on the motivation behind this change. Currently, the Neuropixels API is only available for Windows, so the cross-platform fixes you've implemented won't be effective until the API is also available for Linux and macOS. We're waiting on the release of the Neuropixels API source code, which will allow us to compile the library for these platforms. Until then, the plugin will remain Windows-only.

I even attempted to compile your changes on my Fedora 40 Linux system, but the build fails at the final step. It incorrectly tries to link the Windows-specific Neuropixels library to the plugin. To resolve this, I had to update the CMakeLists.txt [here](https://github.com/open-ephys-plugins/neuropixels-pxi/blob/main/CMakeLists.txt#L104-L105) to only link the libraries when building on Windows. After making that change, the plugin built successfully but failed to load in the GUI because it couldn't find the necessary Neuropixels library symbols.

```
[open-ephys] ***ERROR*** neuropixels-pxi.so: undefined symbol: setGain
[open-ephys] ***ERROR*** neuropixels-pxi.so Load FAILED
```

Could you confirm how you tested this on your end? Please let me know if there's something I'm overlooking.

kontexio commented:
hello @anjaldoshi Thanks for the getting back to us.  We are currently refreshing the Gen2 XDAQs to support Neuropixels and the new implementation will support all three platforms.  This is a courtesy request in case you are interested.

kivancgnlp committed:
include directive moved onto the top because of a symbol name collision in "C" global namespace

np::close() function was colliding with the file systems's close() function in "C" linkage

kivancgnlp committed:
Revert "include directive moved onto the top because of a symbol name collision in "C" global namespace"

This reverts commit 2b2f45767adc0f6d0a7792cd4ceff6d6c399341d.

kivancgnlp committed:
Mock libraries for test purposes

kivancgnlp committed:
+

kivancgnlp committed:
Mock NPAPI Link libraries added for test purposes

kivancgnlp commented:
Hi @anjaldoshi, I've recently tested the OE Plugin on Ubuntu 22 and saw it is working without problem. I've also added mock NPAPI libraries so you can test it without hardware dependency.

jsiegle commented:
Is the plan to create a version of the Neuropixels API that communicates with XDAQ hardware instead of IMEC's basestations? If so, we will probably want to release this as a separate plugin (e.g. "XDAQ Neuropixels"), so users are sure to install the right version for their hardware. We can do that by forking the plugin from the `kontex-neuro` account, similar to how we released the XDAQ plugin for Intan headstages.

kontexio commented:
Yes, we will be releasing a separate plugin, as you suggested.  We are notifying you as a courtesy in case you are ever interested in porting to other platforms.

jsiegle committed:
Quad base updates

jsiegle committed:
Add support for NP2000 and NP2010 IMRO part numbers

jsiegle committed:
Automatically calibrate simulated probe

jsiegle committed:
Reset probe pointers when refreshing plugin

jsiegle committed:
Start canvas acquisition even if probe is disabled

medengineer committed:
Restore missing NP2 PN

jsiegle committed:
Disable quick-load IMRO ComboBox during acquisition

anjaldoshi committed:
Show electrode info in a tooltip window

medengineer committed:
Restore sync state after refresh

medengineer committed:
Update workflow

medengineer committed:
Update to API v3.70.3

jsiegle committed:
Change XML attribute name for quad base

jsiegle committed:
Update plugin version

anjaldoshi committed:
Use correct XML attribute name for quad base when loading settings

jsiegle committed:
Update files for API v4.0.0

jsiegle committed:
Start converting code to API v4.0.0

jsiegle committed:
Finish API v4.0.0 updates

jsiegle committed:
Fix UHD preset selection

jsiegle committed:
Save Quad Base settings as separate probes

jsiegle committed:
Merge branch 'juce8' into apiv4.0.0

jsiegle committed:
Update quad base reloading

jsiegle committed:
Update quad base reloading

anjaldoshi committed:
Update plugin version to 1.0.0

jsiegle committed:
Fix OneBox ADC initialization

jsiegle committed:
Fix OneBox sync initialization

jsiegle committed:
Add option to cancel Neuropixels device scan

jsiegle committed:
Add part numbers and standardize electrode x-offset

anjaldoshi committed:
Refactor GHA workflow

jsiegle committed:
Merge branch 'juce8' into apiv4.0.0

anjaldoshi committed:
Bump plugin version to 1.0.1

jsiegle committed:
Remove call to setWaveform

jsiegle committed:
Increment version number

jsiegle committed:
Add firmware .zip files

jsiegle committed:
Merge pull request #43 from open-ephys-plugins/v1.0.2

- Remove call to `setWaveform`
- Add firmware `.zip` files

StoneLin0708 committed:
Merge remote-tracking branch 'upstream/main'

StoneLin0708 committed:
Drop mock libs

jsiegle committed:
Update to Neuropixels API v4.1.2

StoneLin0708 committed:
Merge remote-tracking branch 'upstream/apiv4.1.2'

StoneLin0708 committed:
Use c++ api

anjaldoshi committed:
Fix channel name in electrode info tooltip

anjaldoshi committed:
Bump plugin version to 1.0.3

jsiegle committed:
Merge pull request #44 from open-ephys-plugins/fix-electrode-tooltip

Fix electrode tooltip

jsiegle committed:
Update to API v4.1.3

jsiegle committed:
Increment plugin version

jsiegle committed:
Update compatible BS firmware versions

jsiegle committed:
Update required firmware version

jsiegle committed:
Add status word parsing for Quad Base

anjaldoshi committed:
Fix path coordinates in UHD Active probe geormetry

jsiegle committed:
Display BSC serial number along with probe info

anjaldoshi committed:
Refactor ProbeBrowser interface

anjaldoshi committed:
Add bank ticks/labels in the Probe Browser zoom area

anjaldoshi committed:
Add electode depth labels in the Probe Browser zoomed-in view

anjaldoshi committed:
Update probe geometry and add depth information in zoomed-out view

jsiegle committed:
Add checks for opto basestation

anjaldoshi committed:
Fix electrode row calculations in `getElectrodesWithinBounds`

jsiegle committed:
Update required firmware version for Opto basestation

jsiegle committed:
Merge branch 'apiv4.1.3' of https://github.com/open-ephys-plugins/neuropixels-pxi into apiv4.1.3

StoneLin0708 committed:
Add XDAQ support

StoneLin0708 committed:
Add npapi package & Fix mac build

StoneLin0708 committed:
CMake cleanup

jsiegle committed:
Allow firmware updates on older basestations

anjaldoshi committed:
Add bandpass filter functionality to ActivityView

anjaldoshi committed:
Move ActivityView implementation from header to source file

anjaldoshi committed:
Calculate ActivityView values only when ProbeBrowser is visible

StoneLin0708 committed:
Rename project & adapt libxdaq api

StoneLin0708 committed:
Add build github workflow

StoneLin0708 committed:
Fix windows build

StoneLin0708 committed:
Fix mac plugin search folder

StoneLin0708 committed:
Add publish workflow

anjaldoshi committed:
Implement CAR functionality in ActivityView

StoneLin0708 committed:
Cache apt packages in Ubuntu workflow

StoneLin0708 committed:
Upgrade np api for np2 timestamp fix

StoneLin0708 committed:
Update visualizer name for XDAQ

anjaldoshi committed:
Adjust legend positioning and color in NeuropixInterface

anjaldoshi committed:
Add SurveyInterface for probe survey functionality

StoneLin0708 committed:
API update for basestation impl

StoneLin0708 committed:
Bump version to 0.1.2

anjaldoshi committed:
Update ActivityView for survey mode and electrode mapping functionality

anjaldoshi committed:
Enhance probe survey run process and update Probe Browser for SurveyInterface

- Introduced SurveyProbePanel class to encapsulate the visualizer for individual probes during surveys.
- Added DisplayMode enum to ProbeBrowser for rendering modes (Interactive, OverviewOnly).
- Updated SurveyInterface to manage multiple SurveyProbePanels and their layout.
- Enhanced SurveyRunner to handle electrode configurations for probes during surveys.

anjaldoshi committed:
Fix probe electrode selection not restoring after survey

anjaldoshi committed:
Update electrode color handling in ProbeBrowser

anjaldoshi committed:
Prevent launching survey while acquisition is active

anjaldoshi committed:
Add functionality to save survey results as JSON

anjaldoshi committed:
Add support for Shank D configurations in NP2 4-shank

anjaldoshi committed:
Fix bank and shank selection in survey interface to show actual button states

anjaldoshi committed:
Add toggle functionality for settings panel in SurveyInterface

anjaldoshi committed:
Add amplitude range selection and legend in SurveyInterface

anjaldoshi committed:
Add recording toggle functionality to SurveyInterface

anjaldoshi committed:
Remove unnecessary survey data reset in `ActivityView::reset` method

anjaldoshi committed:
Reset abstract FIFOs when disabling survey mode in ActivityView

anjaldoshi committed:
Add survey mode-specific colors and max amplitude to ProbeBrowser

anjaldoshi committed:
Reset peak-to-peak values when starting new survey

anjaldoshi committed:
Disable run survey button during non-survey data acquisition

anjaldoshi committed:
Add bandpass filter and CAR toggle buttons to SurveyInterface

anjaldoshi committed:
Replace seconds per bank slider with combo box for time selection in SurveyInterface

anjaldoshi committed:
Add amplitude range combo box to NeuropixInterface

anjaldoshi committed:
Fix maximum peak-to-peak amplitude handling in NeuropixInterface

anjaldoshi committed:
Add dynamic panel width adjustment based on shank count in SurveyInterface

jsiegle committed:
Add electrode index to XML file

jsiegle committed:
Update survey interface layout

jsiegle committed:
Remove debug output

anjaldoshi committed:
Improve survey cancellation handling

anjaldoshi committed:
Remove unnecessary tab index setting during canvas initialization

anjaldoshi committed:
Add parameter save and load functionality to SurveyInterface

anjaldoshi committed:
Update run survey button styling and add play symbol

anjaldoshi committed:
Enable real-time survey panel updates during non-survey acquisition

anjaldoshi committed:
Enhance progress reporting during survey execution

jsiegle committed:
Ensure simulated probes can synchronize

StoneLin0708 committed:
Fix extra headstages after rescan

StoneLin0708 committed:
Temporary disable unimplemented bist

StoneLin0708 committed:
Bump version to 0.1.3

jsiegle committed:
Add shank, xpos, and ypos metadata to channel objects

jsiegle committed:
Fix tooltip duplication

jsiegle committed:
Remove debug output

jsiegle committed:
Print shank status for shift register BIST

jsiegle committed:
Change color of shanks with shift register errors

jsiegle committed:
Retrieve SimulatedProbe event code from static object

jsiegle committed:
Add back start time to simulated probe

StoneLin0708 committed:
Disable sync menu

StoneLin0708 committed:
Update readme for XDAQ

anjaldoshi committed:
Update available banks for NHP2 probe types

anjaldoshi committed:
Disable NP2 HS Led state during initialization

anjaldoshi committed:
Reset `probesInitialized` flag in Basestations to allow re-initialization

anjaldoshi committed:
Fix max peak-to-peak value mismatch

anjaldoshi committed:
Merge branch 'main' into 'refactor-activity-view'

anjaldoshi committed:
Update ProbeInterface annotations to include model name and description

anjaldoshi committed:
Update CMakeLists to only install DLLs from Resources directory

anjaldoshi committed:
Fix ProbeBrowser in survey interface to display electrode index instead of row index

anjaldoshi committed:
Fix legend display to use correct max peak-to-peak amplitude

anjaldoshi committed:
Adjust bounds for firmware update components

anjaldoshi committed:
Fix electrode info string to use 0-based indexing for channels and electrodes

anjaldoshi committed:
Add firmware .zip files

StoneLin0708 committed:
Revert "Temporary disable unimplemented bist"

This reverts commit a30467eff8157319dc282dbfe998f0149db26823.

StoneLin0708 committed:
Update API for np2 unit conversion fixes and linux driver support

StoneLin0708 committed:
Bump version to 0.1.4

StoneLin0708 committed:
Disable github action upload if not a tagged release

anjaldoshi committed:
Add support for Bank D  configurations in NP2_4 SimulatedProbe

anjaldoshi committed:
Update README

anjaldoshi committed:
Fix gain correction for quad base probes

anjaldoshi committed:
Improve calibration file warning message and add hyperlink to CalibrationInfo directory

anjaldoshi committed:
Bump plugin version to 2.0.1

anjaldoshi committed:
Enable paste button based on copied probe settings

anjaldoshi committed:
Add support for surveying UHD2 probes

anjaldoshi committed:
Fix ProbeBrowser for UHD2 probes in SurveyInterface

anjaldoshi committed:
Bump plugin version to 2.0.2

anjaldoshi committed:
Fix paste settings for UHD2 probes

anjaldoshi committed:
Make enable electrodes button invisible for UHD2 probes

anjaldoshi committed:
Merge pull request #48 from open-ephys-plugins/fix-paste-button

Fix paste button crash and add UHD2 probe support in Survey Mode

StoneLin0708 committed:
Detect unsupported headstages and probes

StoneLin0708 committed:
Merge remote-tracking branch 'upstream/main'

StoneLin0708 committed:
XDAQ device timestamp support

StoneLin0708 committed:
Fix msvc build

StoneLin0708 committed:
ci: use conan index project

StoneLin0708 committed:
Upgrade npapi for np2 detection fix

StoneLin0708 committed:
log when unknow headstage detected

anjaldoshi committed:
Improve calibration file handling for all probe types

anjaldoshi committed:
Fix reference selection for NP2 multishank probes with broken shanks

anjaldoshi committed:
Bump plugin version to 2.0.3

anjaldoshi committed:
Add calibration status indicator to NeuropixInterface

anjaldoshi committed:
Add warning for damaged shanks

jsiegle committed:
Update Ext ref selection method

anjaldoshi committed:
Disable damaged shank warning during survey mode

anjaldoshi committed:
Merge pull request #49 from open-ephys-plugins/fix-reference-selection

Fix reference selection for multishank probes and show damaged shank warnings

jsiegle committed:
Fix acquisition with multiple OneBoxes

jsiegle committed:
Prevent slot number conflicts

StoneLin0708 committed:
Add XDAQ ADC

StoneLin0708 committed:
Bump version to 0.3.0

StoneLin0708 committed:
Upgrade api for FW compatibility fix

StoneLin0708 committed:
Fix ADC thread teardown

StoneLin0708 committed:
XDAQ ADC UI

StoneLin0708 committed:
Bump version to 0.3.1

jsiegle committed:
Fix ActivityView channel mapping

jsiegle committed:
Sort channels before writing settings and IMRO files

jsiegle committed:
Ensure SurveyInterface is properly resized on launch

jsiegle committed:
Bump version number to 2.1.0

jsiegle committed:
Remove debug output

anjaldoshi committed:
Update DataStream settings to use stream-specific description and identifier

anjaldoshi committed:
Ensure probe configuration is restored after running shift register test

jsiegle committed:
Re-initialize probes after running BISTs

anjaldoshi committed:
Bump plugin version to 2.1.1

StoneLin0708 committed:
Upgrade api for synclink firmware detection

StoneLin0708 committed:
Bump version to 0.3.2

StoneLin0708 committed:
Upgrade api for Core 2 port detection fix

StoneLin0708 committed:
Bump version to 0.3.3

StoneLin0708 committed:
update libxdaqnp and install interface

StoneLin0708 committed:
run build workflow on push

StoneLin0708 committed:
fix windows package

StoneLin0708 committed:
fix windows streaming

StoneLin0708 committed:
update actions

StoneLin0708 committed:
Bump version to 0.4.0

StoneLin0708 committed:
Fix rescan

StoneLin0708 committed:
use default search path on xdaq scanBS api

StoneLin0708 committed:
Bump version to 0.4.1 & fix np2 config / single shank detection

StoneLin0708 committed:
Prevent double trigger ci build

rylanhsu committed:
Fix plugin loading issue of a fresh install on windows

StoneLin0708 committed:
Fix device discover conflict between XDAQ plugins

StoneLin0708 committed:
Bump version to 0.4.2

StoneLin0708 committed:
Fix library search path on win and linux

StoneLin0708 committed:
Fix np1 timestamp_s buffer size

StoneLin0708 committed:
Merge upstream version 2.1.1

StoneLin0708 committed:
Bump version to 0.5.0

- fix np1 timestamp buffer
- update remote tracking to 2.1.1

StoneLin0708 committed:
Add windows installer build

rylanhsu committed:
fix: correct Windows installer path and split CI for platform-specific packaging (#2)

- Windows Installer Path: Updated CPACK_NSIS_INSTALL_ROOT in CMakeLists.txt to $PROFILE\AppData\Local
- CI/CD Pipeline Restructuring: Split the monolithic Linux publish step into three distinct jobs for running OS-specific commands.
- Windows Packaging Fix: Added a "Repack Package" step to fix the plugin packaging issue that was occurring on Windows.

StoneLin0708 committed:
Bump version to 0.5.1

StoneLin0708 committed:
Fix ci task dependency
